File: app/src/main/java/com/bis5/fitjourney/models/ExerciseDao.java
package com.bis5.fitjourney.models;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ExerciseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Exercise exercise);

    // OBEY THE ONE TRUE LAW: The workoutId is a long.
    @Query("SELECT * FROM exercises WHERE workoutId = :workoutId ORDER BY id ASC")
    LiveData<List<Exercise>> getExercisesForWorkout(long workoutId);

    @Query("DELETE FROM exercises WHERE id = :exerciseId")
    void deleteExercise(int exerciseId);

}

File: app/src/main/java/com/bis5/fitjourney/models/ExerciseDao.kt
package com.bis5.fitjourney.models

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface ExerciseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertExercise(exercise: Exercise)

    @Query("SELECT * FROM exercises WHERE workoutId = :workoutId ORDER BY id ASC")
    fun getExercisesForWorkout(workoutId: String): Flow<List<Exercise>>

    @Query("DELETE FROM exercises WHERE id = :exerciseId")
    suspend fun deleteExercise(exerciseId: Int)

}